client/src/client-only-routes/ShowProjectLinks.js

```diff
@@ -0,0 +1,167 @@
+import { projectMap } from '../resources/certAndProjectMap';
+import { Link } from 'gatsby';
+import ProjectModal from '../components/SolutionViewer/ProjectModal';
+import { find } from 'lodash';
+
+const propTypes = {
+  certName: PropTypes.string,
+  name: PropTypes.string,
+  user: PropTypes.shape({
+    completedChallenges: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        solution: PropTypes.string,
+        githubLink: PropTypes.string,
+        files: PropTypes.arrayOf(
+          PropTypes.shape({
+            contents: PropTypes.string,
+            ext: PropTypes.string,
+            key: PropTypes.string,
+            name: PropTypes.string,
+            path: PropTypes.string
+          })
+        )
+      })
+    ),
+    username: PropTypes.string
+  }),
+  username: PropTypes.string
+};
+
+const initSolutionState = {
+  projectTitle: '',
+  files: null,
+  solution: null,
+  isOpen: false
+};
+
+const ShowProjectLinks = props => {
+  const [solutionState, setSolutionState] = useState(initSolutionState);
+
+  const handleSolutionModalHide = () => setSolutionState(initSolutionState);
+
+  const getProjectSolution = (projectId, projectTitle) => {
+    const {
+      user: { completedChallenges }
+    } = props;
+
+    const completedProject = find(
+      completedChallenges,
+      ({ id }) => projectId === id
+    );
+
+    if (!completedProject) {
+      return null;
+    }
+
+    const { solution, githubLink, files } = completedProject;
+    const onClickHandler = () =>
+      setSolutionState({
+        projectTitle,
+        files,
+        solution,
+        isOpen: true
+      });
+
+    if (files && files.length) {
+      return (
+        <button
+          onClick={onClickHandler}
+          className='project-link-button-override'
+        >
+          solution
+        </button>
+      );
+    }
+    if (githubLink) {
+      return (
+        <>
+          <a href={solution} rel='noopener noreferrer' target='_blank'>
+            solution
+          </a>
+          ,{' '}
+          <a href={githubLink} rel='noopener noreferrer' target='_blank'>
+            source
+          </a>
+        </>
+      );
+    }
+    if (maybeUrlRE.test(solution)) {
+      return (
+        <a
+          block={'true'}
+          className='btn-invert'
+          href={solution}
+          rel='noopener noreferrer'
+          target='_blank'
+        >
+          solution
+        </a>
+      );
+    }
+    return (
+      <button className='project-link-button-override' onClick={onClickHandler}>
+        solution
+      </button>
+    );
+  };
+
+  const renderCertification = certName => (
+    <ul key={certName}>{renderProjectsFor(certName)}</ul>
+  );
+
+  const renderProjectsFor = certName => {
+    return projectMap[certName].map(({ link, title, id }) => (
+      <li key={id}>
+        <Link to={link}>{title}</Link>: {getProjectSolution(id, title)}
+      </li>
+    ));
+  };
```

Comment:
```suggestion
  const renderProjectsFor = certName =>
    (projectMap[certName] || legacyProjectMap[certName]).map(
      ({ link, title, id }) => (
        <li key={id}>
          <Link to={link}>{title}</Link>: {getProjectSolution(id, title)}
        </li>
      )
    );
```
This handles legacy certs.

I'm not sure what's the best way to handle a weird bug where the cert renders, but is not in either of the Maps.  Maybe show

> Something has gone wrong with this certification, please report this to our team

@ahmadabdolsaheb any thoughts?

Reply:
This was a silly oversight of mine. Thanks, Oliver.

The only cert currently not in the map is the Fullstack cert. Once. #39695 is complete, then it should be easy enough to add. Should I push something _dirty_ (hardcoded) through, for this PR?